Completing the using branches how to guide (#191)
neondatabase/website: content/docs/how-to-guides/using-branches.md

---
title: How to Use Branches
---

> Now that you have access to [branching](https://neon.tech/docs/conceptual-guides/branching.md) on Neon, let's look at how you can use it.

Reviewer: let's add a placeholder for those who discover this page without a having a branching permission
users can get early access to branching by sending an email and describing their use case

> Branching allows you to create a copy of your data so that you can modify a version of your project data without changing your production data.

Reviewer 2: ```suggestion
Branching allows you to create a copy of your Project data which you can modify without affecting the data you have in production.
```

> A new branch has all the same data that your project had when you first created the branch, but future changes to the data in your project won't update your branch.

Reviewer 2: ```suggestion
A branch is created with the same data that existed in your Project at the time the branch was created, but future changes to your Project data do not affect the branch.
```

> Below is a guide to using branching in your project.

Reviewer 2: ```suggestion
The following guide describes how to using branching in your project.
```

> Branches currently appear as a whole new project in the Neon UI and API, but when branches become available to all users, you'll be able to find them within their parent project.

Reviewer 2: ```suggestion
Branches currently appear as a new project in the Neon Console and API, but when the branching feature is made available to all users, branches will be associated with their parent Project.
```

---

> Let's look at how you can create branches using either the Neon Dashboard or the Neon API.

Reviewer 2: ```suggestion
Let's look at how to create branches using the Neon Console and the Neon API.
```

> Create a Project on Neon and Add Your Data

Reviewer 2: ```suggestion
Create a Project in Neon and add your data
```

===============

> Log in to Neon and create a project on Neon in the UI: https://console.neon.tech

Reviewer 2: ```suggestion
Log in to Neon and create a project in the Neon Console: [https://console.neon.tech](https://console.neon.tech)
```

> Add some data using the SQL Editor in the UI, psql, or any other Postgres client.

Reviewer 2: ```suggestion
Add data to your Project using the Neon SQL Editor, `psql`, or some other PostgreSQL client.
```

> Our docs have more information about how to create a project :

Reviewer 2: ```suggestion
Refer to the Neon documentation for information about creating a Project :
```

https://neon.tech/docs/cloud/getting-started/
https://neon.tech/docs/cloud/tutorials/
https://neon.tech/docs/quickstart/postgres/

> 

Reviewer 2: Maybe just reference this topic:
Setting up a Project
https://neon.tech/docs/get-started-with-neon/setting-up-a-project/

> Once you have a project with data in it, you can click the `Create branch` button in your project Dashboard to finish creating your branch.

Reviewer 2: ```suggestion
After creating a Project and adding data, click **Create branch** on the **Dashboard** tab in the Neon Console to create a branch.
```

> Alternatively, you can use the Neon API to create your branch.

Reviewer 2: ```suggestion
Alternatively, you can use the Neon API to create a branch.
```

> Create a Branch Using the API

Reviewer 2: Create a branch using the API

======================================

> Get familiar with using API keys on Neon: https://neon.tech/docs/get-started-with-neon/using-api-keys/

Reviewer 2: ```suggestion
Using the Neon API requires an API key. For information about how to obtain an API key for your Project, refer to [Using API keys](https://neon.tech/docs/get-started-with-neon/using-api-keys/).
```

> The Neon API Reference: https://console.neon.tech/api-docs

Reviewer 2: ```suggestion
For information about the API methods that Neon supports, refer to the [Neon API Reference](https://console.neon.tech/api-docs). 
```

> To create a branch, you can use the API method:

Reviewer 2: ```suggestion
To create a branch, use the following API method:
```

	POST /projects/{project_id}/branches

Here's an example of a curl command that you can use to create a branch: 

```
curl -o - -X POST -H 'Authorization: Bearer ...' https://console.neon.tech/api/v1/clusters/ancient-haze-985396/branches
```

> To use this curl command, change the placeholder project name `ancient-haze-985396` to the name of the project that you want to create a branch from.

Reviewer 2: ```suggestion
To use the cURL command with your Project, change the placeholder Project name `ancient-haze-985396` to the name of the Project that you want to branch from.
```

In the response, you will receive new project info info, with the name that looks like
`$parent_project_id-branch-...`.

> Currently branches appear as a separate project in the Dashboard UI, this will change in the future when branches are publically available.

Reviewer 2: ```suggestion
Currently, a branch appears as separate Project on the **Dashboard** tab in the Neon Console. This will change in the future when the branching feature becomes generally available.
```

All the data in the parent project when you create the branch will be available in the branched project as well.
Future changes to the data in the parent and the branch are independent, so you can make changes to the branch without affecting your parent project.

> Learn More About [Branching](https://neon.tech/docs/conceptual-guides/branching.md) in Neon

Reviewer 2: ```suggestion
To learn more about Neon's branching feature, see [Branching](https://neon.tech/docs/conceptual-guides/branching.md), in the Neon documentation.
```
